builtins-default-costs/src/lib.rs

```diff
@@ -25,108 +19,35 @@ struct BuiltinCost {
 }
 
 lazy_static! {
-    /// Number of compute units for each built-in programs
-    ///
-    /// DEVELOPER WARNING: This map CANNOT be modified without causing a
-    /// consensus failure because this map is used to calculate the compute
-    /// limit for transactions that don't specify a compute limit themselves as
-    /// of https://github.com/anza-xyz/agave/issues/2212.  It's also used to
-    /// calculate the cost of a transaction which is used in replay to enforce
-    /// block cost limits as of
-    /// https://github.com/solana-labs/solana/issues/29595.
```

Comment:
Removing the warning isn't going to make this issue go away. We can't update this static map without impacting the  calculated default compute unit limit that we reserve transactions. Any changes need to be feature gated